src/schnorrsig.rs

```diff
@@ -449,6 +450,38 @@ impl<'de> ::serde::Deserialize<'de> for PublicKey {
     }
 }
 
+impl SecretKey {
+    /// Creates a new secret key using data from BIP-340 [`KeyPair`]
+    pub fn from_keypair<V: Verification>(secp: &Secp256k1<V>, keypair: &KeyPair) -> Self {
```

Comment:
This is wrong, `secp256k1_keypair_sec` doesn't require the context to be initialized for Verification, it can use the no precomp static context here

Reply:
Fixed in #306. https://github.com/rust-bitcoin/rust-secp256k1/pull/308/commits/c241c570a3114d5344b76c380402e0a61f56e130 

BTW, what is the overall rule where to use which context?